hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/MessageRunner.java

```diff
@@ -93,14 +92,24 @@ public void next() {
 
     @Override
     public void accept(ReadResultSet<ReliableTopicMessage> result, Throwable throwable) {
+        if (cancelled) {
+            return;
+        }
+
         if (throwable == null) {
             // we process all messages in batch. So we don't release the thread and reschedule ourselves;
             // but we'll process whatever was received in 1 go.
-            for (Object item : result) {
-                ReliableTopicMessage message = (ReliableTopicMessage) item;
+            for (int i = 0; i < result.size(); i++) {
+                ReliableTopicMessage message = result.get(i);
 
-                if (cancelled) {
-                    return;
+                long actualSequence = result.getSequence(i);
```

Comment:
Not the topic of this PR but I can see how this `MessageRunner` cannot run with a predicate since it expects a contiguous range of sequences.

Reply:
We use a different formula in event journal, I think it could be used here too? 
`lostCount = resultSet.getNextSequenceToReadFrom() - resultSet.readCount() - prevSequence`, where prevSequence is the asked for sequence.

Reply:
Yes, we could report this in the log message below.

Reply:
I'm not 100% sure what you need me to do about it, so pls. check what I'll push.

Reply:
Looks good :)